hprotoc/Text/ProtocolBuffers/ProtoCompile/Gen.hs

```diff
@@ -786,6 +807,96 @@ instanceUnknownMessage di
         ]
   where putunknownfield = RecUpdate () (lvar "msg") [ FieldUpdate () (localField di "unknown'field") (lvar "u'f") ]
 
+instanceToJSON :: DescriptorInfo -> Decl ()
+instanceToJSON di
+  = InstDecl () Nothing (mkSimpleIRule (private "ToJSON") [TyCon () (unqualName (descName di))]) . Just $
+      [ inst "toJSON" [patvar msgVar] serializeFun
+      ]
+  where
+        name = mName $ baseName $ descName di
+        flds = F.toList (fields di)
+        msgVar = distinctVar "msg"
+        reservedVars = map toPrintName flds
+        distinctVar var = if var `elem` reservedVars then distinctVar (var ++ "'") else var
+        getFname fld = fName $ baseName' $ fieldName fld
+        makePair fld =
+            let fldName = getFname fld
+                fldName' = dropWhileEnd (== '\'') fldName
+                toJSONFun = case typeCode fld of
+                    3 -> pvar "toJSONShowWithPayload"
```

Comment:
Gibt's für diese Zahlen keine symbolischen Konstanten??

Reply:
Yes, but I thought I couldn't use them. There is http://hackage.haskell.org/package/protocol-buffers-descriptor-2.4.10/docs/Text-DescriptorProtos-FieldDescriptorProto-Type.html#t:Type, but that would mean parsing it first.

Reply:
what about introducing symbolic constants?

Reply:
I just added the import. The package was in scope anyway. Not sure what this does for bootstrapping though.